src/Idephix/Console/Command.php

```diff
@@ -1,61 +1,126 @@
+use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Input\InputOption;
-use Symfony\Component\Console\Input\InputArgument;
-use Idephix\Util\DocBlockParser;
 use Symfony\Component\Console\Output\OutputInterface;
 
-class CommandWrapper extends Command
+class Command extends SymfonyCommand
 {
-    private $idxTask;
-
+    private $idxTaskCode;
+    /** @var  IdephixInterface */
     private $idx;
 
-    public function withIdx(IdephixInterface $idx)
-    {
-        $this->idx = $idx;
-
-        return $this;
-    }
-
     /**
+     * Build a command from callable code
+     *
+     * This is maintained only to support legacy idxfile and will soon removed
+     *
      * @param callable $code
      * @return $this
+     * @deprecated
      */
-    public function buildFromCode($code)
+    public static function buildFromCode($name, $code, IdephixInterface $idx)
     {
-        $this->assertCallable($code);
-        $this->idxTask = $code;
+        $command = new static($name);
+        $command->idx = $idx;
+
+        $command->assertCallable($code);
+        $command->idxTaskCode = $code;
 
         $reflector = new \ReflectionFunction($code);
         $parser = new DocBlockParser($reflector->getDocComment());
-        $this->setDescription($parser->getDescription());
+        $command->setDescription($parser->getDescription());
 
         foreach ($reflector->getParameters() as $parameter) {
             $description = $parser->getParamDescription($parameter->getName());
 
             if ($parameter->getName() !== 'idx') {
-                $this->addParameter($parameter, $description);
+                $command->addParameter($parameter, $description);
             }
         }
 
-        return $this;
+        return $command;
+    }
+
+    /**
+     * @param Task $task
+     * @return Command
+     */
+    public static function fromTask(Task $task, IdephixInterface $idx)
+    {
+        $command = new static($task->name());
+        $command->idx = $idx;
+
+        $command->setDescription($task->description());
+
+        /** @var Parameter $parameter */
+        foreach ($task->parameters() as $parameter) {
+            if (!$parameter->isOptional()) {
+                $command->addArgument($parameter->name(), InputArgument::REQUIRED, $parameter->description());
+                continue;
+            }
+
+            if ($parameter->isFlagOption()) {
+                $command->addOption($parameter->name(), null, InputOption::VALUE_NONE, $parameter->description());
+                continue;
+            }
+
+            $command->addArgument(
+                $parameter->name(),
+                InputArgument::OPTIONAL,
+                $parameter->description(),
+                $parameter->defaultValue()
+            );
+        }
+
+        $command->assertCallable($task->code());
```

Comment:
il fatto che sia una callable non è già garantito in fase di costruzione del task?